Add a flow generator function

Eviatar109 opened the pull request:
Adding a function to generate flows,
 making adjustments according to a path file that will work for everyone.
activating the function for each net.
Currently there is a file called route_xml_path_intersection_{num} that contains all the path of the routes files.

Eviatar109 committed:
Add a flow generator function

Eviatar109 committed:
Add a net, rou and sumocfg for each intersection

Eviatar109 committed:
Running the function for each intersection

Eviatar109 committed:
change the name files for the txt files

TalMekler commented on flow_generator.py:
```diff
@@ -0,0 +1,132 @@
+import shutil as sh
+
+
+def add_vehicle_flows(original_xml_file_path, text_xml_file, inset_flows, new_index, level):
```
add parameters type and return type
Example: def add_vehicle_flows(... , new_index: int, level: str) -> None

TalMekler commented on flow_generator.py:
```diff
@@ -0,0 +1,132 @@
+        f.write(f'{new_xml_file_path}\n')
+
+
+def get_routes(xml_file):
```
add parameters type and return type

TalMekler commented on flow_generator.py:
```diff
@@ -0,0 +1,132 @@
+    return routes_list
+
+
+def generate_new_flows(routes, low, high):
```
add parameters type and return type

TalMekler commented on flow_generator.py:
```diff
@@ -0,0 +1,132 @@
+    """
+    new_flows = []
+    for i, route_id in enumerate(routes):
+        vehs_per_hour = random.randint(low, high)
```
maybe add min and max value?

Eviatar109 committed:
Add a type request to the functions

Eviatar109 committed:
change the net in intersection_1 and activate the function

Eviatar109 committed:
change the net in intersection_1 and activate the function

Eviatar109 committed:
Add a type hints

Eviatar109 committed:
Add a method random_flow

insert the whole to loop

Eviatar109 committed:
activate the function for the whole intersection

Eviatar109 committed:
change the flow scale for the different level
activate the function for the whole intersection

Eviatar109 committed:
split between intersection 5 to 6

Eviatar109 committed:
split between intersection 5 to 6

Eviatar109 committed:
Add: complete rou file for full net net file and sumocfg file

Eviatar109 committed:
fix: flow_generator.py

Eviatar109 committed:
Add: trip info for whole intersection unless full net
fix: Passing through all the intersection and making relevant repairs, all the intersections now work
